packages/uni_ui/lib/cards/schedule_card.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:flutter/material.dart';
+import 'package:phosphor_flutter/phosphor_flutter.dart';
+import 'package:uni_ui/cards/generic_card.dart';
+import 'package:uni_ui/theme.dart';
+
+class ScheduleCard extends GenericCard {
+  const ScheduleCard(
+      {super.key,
+      required this.name,
+      required this.acronym,
+      required this.room,
+      required this.type,
+      // required this.badgeColor,
+      this.isActive = false,
+      this.teacherName,
+      this.teacherPhoto});
+
+  final String name;
+  final String acronym;
+  final String room;
+  final String type;
+  // final BadgeColors badgeColor;
+  final bool isActive;
+  final String? teacherName;
+  final String? teacherPhoto;
+
+  @override
+  Widget build(BuildContext context) {
+    return GenericCard(
+      key: key,
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: [
+          Expanded(
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                Row(
+                  children: [
+                    if (isActive) ...[
+                      PhosphorIcon(
+                        PhosphorIcons.clock(PhosphorIconsStyle.duotone),
+                        color: Theme.of(context).colorScheme.secondary,
+                        size: 20,
+                      ),
+                      SizedBox(width: 5),
+                    ],
+                    Text(
+                      acronym,
+                      overflow: TextOverflow.ellipsis,
+                      style: TextStyle(
+                          fontSize: Theme.of(context)
+                              .textTheme
+                              .headlineMedium!
+                              .fontSize,
+                          fontWeight: Theme.of(context)
+                              .textTheme
+                              .headlineMedium!
+                              .fontWeight,
+                          color: isActive
+                              ? Theme.of(context).colorScheme.secondary
+                              : Theme.of(context).colorScheme.primary),
```

Comment:
This is not so elegant. You should assign the theme style directly to the text and not each section. The primary color should be assigned as the color of the text style already

Reply:
Being solved by the redesign team